### PR TITLE
[TT-7275] document external OAuth JWT validation.

### PR DESCRIPTION
**Pull request by jeffy-mathew**
[changelog]
internal: document external OAuth JWT validation.

## Related Issue
https://tyktech.atlassian.net/browse/TT-7275

### CODE REVIEW

**Review by titpetric — APPROVED** [review]

File: apidef/oas/security.go [hunk 1]
```diff
@@ -401,12 +401,22 @@ type OAuthProvider struct {
...
+	// Source is the secret to verify signature, it could be one among
+	// - a base64 encoded static secret
+	// - a valid JWK url in plain text
+	// - a valid JWK url in base64 encoded format.
+	Source string `bson:"source" json:"source"`
+	// IdentityBaseField is the identity claim name.
+	IdentityBaseField string `bson:"identityBaseField,omitempty" json:"identityBaseField,omitempty"`
+	// IssuedAtValidationSkew is the clock skew to be considered while validating iat claim
+	IssuedAtValidationSkew uint64 `bson:"issuedAtValidationSkew,omitempty" json:"issuedAtValidationSkew,omitempty"`
+	// NotBeforeValidationSkew is the clock skew to be considered while validating nbf claim
```
Comment (furkansenharputlu) [inline apidef/oas/security.go:417]: Let’s finish with `.` as godoc convention

Comment (furkansenharputlu) [reply apidef/oas/security.go:417]: Same for others

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4488/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3627398363)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4488/merge
Commit: 343fd7257e01aef6f89fbff2a848faa2b9ce8242 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3694628711)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4488/merge
Commit: fce7132707ea306340c6e7733916bf34ea91e31b 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3694641653)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4488/merge
Commit: 12cf2658dbdf36151fc5c6b9f914bc2a8d525ff8 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3694692108)

**Review by furkansenharputlu — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 2b14ec5ff3a1f2023591145a4fc09a67ea694a86 [TT-7275] document external OAuth JWT validation. (#4488)

[changelog]
internal: document external OAuth JWT validation.

## Related Issue
https://tyktech.atlassian.net/browse/TT-7275
Triggered by: push (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3696029453)

**Pull request merged**